pkg/reconciler/dns/dns_record.go

```diff
@@ -54,6 +55,16 @@ func (c *Controller) reconcile(ctx context.Context, dnsRecord *v1.DNSRecord) err
 		}
 	}
 
+	ingress, err := c.getIngressOwner(ctx, dnsRecord)
+	if err != nil {
+		return err
+	}
+
+	if err = c.ReconcileHealthChecks(ctx, ingress, dnsRecord); err != nil {
+		klog.Errorf("Failed to reconcile health check for DNSRecord %s: %v", dnsRecord.Name, err)
+		return err
+	}
+
```

Comment:
I think i f possible we should avoid the dns record needing to know anything about ingresses. We should think of them as 2 separate pieces entirely, especially since we will be switching to using external-dns very soon and a lot of this logic will go away.

Instead of looking up the ingress to get the annotation, we should just add/remove the annotation to/from the DNSRecord as part of the ingress reconciliation. It will be easier to test, and will also work when we come to use the external dns DNSEndpoint resource.

```suggestion
	if err := c.ReconcileHealthChecks(ctx, dnsRecord); err != nil {
		klog.Errorf("Failed to reconcile health check for DNSRecord %s: %v", dnsRecord.Name, err)
		return err
	}
```